src/main/java/it/eng/idsa/dataapp/web/rest/DataControllerBodyBinary.java

```diff
@@ -52,50 +54,30 @@ public ResponseEntity<?> routerBinary(@RequestHeader HttpHeaders httpHeaders,
 		} else {
 			logger.info("Payload is empty");
 		}
-		
+
 		Message message = MultipartMessageProcessor.getMessage(headerMessage);
 
-		Message headerResponse = messageUtil.getResponseHeader(message);
-		String responsePayload = null;
-		
-		if (!(headerResponse instanceof RejectionMessage)) {
-			responsePayload = messageUtil.createResponsePayload(message, payload);
-		} 
-		if(responsePayload == null && message instanceof ContractRequestMessage) {
-			logger.info("Creating rejection message since contract agreement was not found");
-			headerResponse = messageUtil.createRejectionNotFound(message);
-		}	
-		
-		if(responsePayload == null && message instanceof DescriptionRequestMessage) {
-			logger.info("Creating rejection message since self description could not be fetched");
-			headerResponse = messageUtil.createRejectionInternalRecipientError(message);
-		}
-		
-		if (responsePayload != null && responsePayload.contains("ids:rejectionReason")) {
-			headerResponse = MultipartMessageProcessor.getMessage(responsePayload);
-			responsePayload = null;
-		}
+		// Create handler based on type of message and get map with header and payload
+		DataAppMessageHandler handler = factory.createMessageHandler(message.getClass());
+		Map<String, Object> responseMap = handler.handleMessage(message, payload);
 
 		ContentType payloadContentType = ContentType.TEXT_PLAIN;
-		
-		if(responsePayload != null && messageUtil.isValidJSON(responsePayload)) {
+
```

Comment:
I know that this is old logic, but what happens with payloadContenType if there is no payload?
Also should be checked in other controllers